tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md

```diff
@@ -0,0 +1,209 @@
+# Signature Gating extension
+
+This library is dedicated to provide a mecanism for off-chain validation of a contract invocation. It also involves a `calldata` structure representing contract invocation which can be processed or dispatched to other contracts.
+
+## General idea
+
+The user wants to invoke a contract with an off-chain signed message. The message is signed by a ReferenceSigner (which can be an implicit account or a contract that provides one) which has the responsability to verify if the user wallet is authorized (i.e. via a KYC authentication service). Then the off-chain signed message can be sent as parameter during the contract invocation.
+
+### Replay attack and expiration
+
+In order to prevent that a signature is used twice , the payload must incapsulate a "nonce" field.
+Once a user has received a signed message, he must consume the offchain signed message before an expiration (block level). The expiration has been introduced in case the user lose his authorization.
+
+### Forging a signature
+
+The request of a user (i.e. a contract invocation) is declared as a payload sent to a ReferenceSigner. The ReferenceSigner retrieves the nonce for the user (from contract storage), specifies the expiration (block level) and compact all using a keccak hashing and finally signs this "keccaked payload". This off-chain signature is produced by the ReferenceSigner.
+Then the produced signature and extra data (expiration, nonce, public key) are sent back to the user. The user can now send his transaction (which contains a `txAuthData` (payload + signature + public key + nonce + expiration)) to the smart contract that implements the signature verification mechanism.
+
+![](./pictures/nexera%20global%20workflow.png)
+
+In this case the transaction fee is paid by the user.
+
+The transaction could also be executed by an operator on behalf of the user. In this case the transaction fee is paid by the operator.
+
+### off-chain signture workflow
+
+For a given calldata, the user receives from Nexera
```

Comment:
So the idea here is to write the general case, right? So maybe don't mention Nexera and say "The Signer" or "The Auth Provider" or "The KYC Provider"